inspirehep/modules/records/tasks.py

```diff
@@ -136,26 +136,28 @@ def _get_record(recid):
         pid = PersistentIdentifier.query.filter_by(pid_value=str(recid)).one()
         return get_db_record(pid.pid_type, recid)
 
-    records = get_records_to_merge()
+    records = get_enriched_records()
 
     with db.session.begin_nested():
         for record in records:
             recid = record['control_number']
-            other_recid = get_recid_from_ref(record['new_record'])
-            other_record = _get_record(other_recid)
 
-            logger.info('Merged records: %d and %d', recid, other_recid)
-            record.merge(other_record)
-            record.commit()
+            deleted_ids = [get_recid_from_ref(ref) for ref in record['deleted_records']]
+
+            for deleted_id in deleted_ids:
+                rec_to_merge = _get_record(deleted_id)
```

Comment:
here the problem is that there isn't necessarily a record corresponding to this `recid`, because some deleted records don't migrate properly. If there is, this should be fine, but otherwise, you should register the PID for redirection without there being an underlying record.